crypto/evp/evp_fetch.c

```diff
@@ -444,11 +444,18 @@ const char *evp_first_name(OSSL_PROVIDER *prov, int name_id)
     return ossl_namemap_num2name(namemap, name_id, 0);
 }
 
-int evp_is_a(OSSL_PROVIDER *prov, int number, const char *name)
+int evp_is_a(OSSL_PROVIDER *prov, int number,
+             const char *legacy_name, const char *name)
 {
+    /*
+     * For a |prov| that is NULL, the library context will be NULL, which
+     * is fine, because that's the only place legacy stuff is relevant.
```

Comment:
I don't think this statement is true. I fully expect "legacy" ciphers/digests to be used in conjunction with non-default OPENSSL_CTXs. AFAIK there is nothing in the documentation suggesting you shouldn't be allowed to do that.

Reply:
Er...  so how do you associate legacy cipher/digests with a non-default OPENSSL_CTX?

Reply:
I'm at a loss what to think or how to comment.  Ideas?

Reply:
For everyone's information, @mattcaswell and I have a private chat where we concluded that this is not ideal, but ideal is hard to come by, so we're resorting to accepting this situation for legacy ciphers and just making sure it's documented.

Done